client/state/happychat/middleware.js

```diff
@@ -0,0 +1,34 @@
+
+/**
+ * Internal dependencies
+ */
+import {
+	HAPPYCHAT_TRANSCRIPT_REQUEST,
+} from 'state/action-types';
+
+import { connection } from './common';
+
+import { receiveChatTranscript } from './actions';
+import { getHappychatTranscriptTimestamp } from './selectors';
+
+const debug = require( 'debug' )( 'calypso:happychat:actions' );
+
+export const requestTranscript = ( { getState, dispatch } ) => {
```

Comment:
Good point! I've modified it a little to take the connection as a parameter, and added a unit test in fcdcf44116a8b9b5b64ffd20e4450001f7a7215c